### DISCUSSION TITLE
[Question] Which units should be used?

### DISCUSSION THREAD

**Discussion by Martin15135215** [adtzlr/trusspy#58] [Q&A] [answered]
Which units should be used for:

- Length: m ?
-  Area: m² ?
- Force: kN ?
- E-modulus: kN/m² ?
- Degree: rad ?

So what units should the numbers I write in the functions have?

**Comment by adtzlr** [comment, marked as answer]
Hi @Martin15135215 and thanks for asking! You are free to use any consistent unit system you like, e.g. one of https://www.dynasupport.com/howtos/general/consistent-units. TrussPy has no units or unit-converters "built-in". SI-Units are always fine. In TrussPy, there are no rotations involved, just (translational) displacements.

P.S.: Your suggestion is also fine. Just be consistent with length and force units over geometric (length, area), kinematic (displacements), constitutive (elastic modulus) and kinetic (force) quantities. The unit system restriction is somewhat relaxed because we have no dynamics and no gravity/mass (sometimes also called body) loads involved in TrussPy.
